robottelo/cli/base.py

```diff
@@ -374,6 +374,13 @@ def _construct_command(cls, options=None):
                 if val is True:
                     tail += u' --{0}'.format(key)
                 elif val is not False:
+                    if isinstance(val, list):
+                        val = ','.join(
+                            str(el.get('id') or el.get('name'))
```

Comment:
The idea here is following:
1) if user passed a list of names/ids/labels/etc - then make a string of them and proceed.
2) but, if user was lazy enough not to send the list of desired fields, but sent a list of entities (in cli they're not separate entity class, but just dictionaries) - then at least try to generate string using most common entity fields - `ID` or, if id not found, `name`.
As in most cases we use IDs - it allows us to simplify from this:
```python
result = Location.update({
           'id': loc['id'],
           'hostgroups': [hg['id'] for hg in new_host_groups],
       })
```
to this:
```python
result = Location.update({
           'id': loc['id'],
           'hostgroups': new_host_groups,
       })
```